google-cloud-firestore/src/main/java/com/google/cloud/firestore/UpdateBuilder.java

```diff
@@ -116,34 +109,30 @@ public T create(
 
   private T performCreate(
       @Nonnull DocumentReference documentReference, @Nonnull Map<String, Object> fields) {
+    verifyNotCommitted();
     DocumentSnapshot documentSnapshot =
         DocumentSnapshot.fromObject(
             firestore, documentReference, fields, UserDataConverter.NO_DELETES);
     DocumentTransform documentTransform =
         DocumentTransform.fromFieldPathMap(
             documentReference, convertToFieldPaths(fields, /* splitOnDots= */ false));
 
-    Mutation mutation = addMutation();
-    mutation.precondition = Precondition.exists(false).toPb();
-
-    if (!documentSnapshot.isEmpty() || documentTransform.isEmpty()) {
-      mutation.document = documentSnapshot.toPb();
-    }
+    Write.Builder write = documentSnapshot.toPb();
+    write.setCurrentDocument(Precondition.exists(false).toPb());
 
     if (!documentTransform.isEmpty()) {
-      mutation.transform = documentTransform.toPb();
+      write.addAllUpdateTransforms(documentTransform.toPb());
     }
 
+    writes.add(write);
+
     return (T) this;
   }
 
-  /** Adds a new mutation to the batch. */
-  private Mutation addMutation() {
+  /** Adds a new write to the batch. */
```

Comment:
Incorrect javadoc now that the mutation has been removed.

Reply:
Removed the javadoc since the method is pretty straightforward now.